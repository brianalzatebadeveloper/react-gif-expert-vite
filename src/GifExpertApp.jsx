import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

  const [ categories, setCategories ] = useState([]);

  const onAddCategory = (value) => {

    if ( !categories.includes(value) )
    {
      setCategories(ctg => [value, ...ctg]);
    }
  }

  return (
    <>
      {/* title */}
      <h1>GifExpertApp</h1>

      {/* input */}
      <AddCategory onAddCategory={onAddCategory} />

      {/* list gifs */}
      <ol>
        { 
          categories.map( category => (
              <GifGrid 
                key={category}
                category={category} 
              />
          )) 
        }
      </ol>

      {/* gif item */}
    </>
  )
}
