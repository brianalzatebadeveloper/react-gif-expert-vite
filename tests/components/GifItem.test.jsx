import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('Pruebas del componente <GifItem />', () => {

  const image = {
    id: '3412341234dsadf',
    title: 'Can You Smell The Rock GIF by WWE',
    url: 'https://media4.giphy.com/media/l0HUg6Ypas42ubkXu/giphy.gif?cid=0858df19h0fqpsnq8bllqgb7kq216cdk3hhg8dv7ah5vgkmb&rid=giphy.gif&ct=g'
  };

  test('debe hacer match con el snapshot', () => {
    const { container } = render( <GifItem { ...image } /> );
    expect( container ).toMatchSnapshot();
  });

  test('debe mostrar la imagen con la URL y el ALT indicado', () => {
    render( <GifItem { ...image } /> );
    const { src, alt } =  screen.getByRole('img');
    expect( src ).toBe( image.url );
    expect( alt ).toBe( image.title );
  });
  
  test('debe mostrar el titulo en el componente', () => {
    render( <GifItem { ...image } /> );
    expect( screen.getByText( image.title ) ).toBeTruthy();
  });

});