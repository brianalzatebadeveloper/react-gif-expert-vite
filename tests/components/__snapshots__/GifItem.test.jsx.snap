// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Can You Smell The Rock GIF by WWE"
      src="https://media4.giphy.com/media/l0HUg6Ypas42ubkXu/giphy.gif?cid=0858df19h0fqpsnq8bllqgb7kq216cdk3hhg8dv7ah5vgkmb&rid=giphy.gif&ct=g"
    />
    <p>
      Can You Smell The Rock GIF by WWE
    </p>
  </div>
</div>
`;
